src/main/kotlin/com/outsystems/plugins/barcode/view/OSBARCScannerActivity.kt

```diff
@@ -394,15 +397,21 @@ class OSBARCScannerActivity : ComponentActivity() {
                 var rectHeight: Float
 
                 if (isPhone) { // for phones
-                    rectWidth = canvasWidth - (horizontalPadding.toPx() * 2) - (ScannerAimRectCornerPadding.toPx() * 2)
-                    rectHeight = canvasHeight - (verticalPadding.toPx() * 2) - (ScannerAimRectCornerPadding.toPx() * 2)
-                } else { // for tablets
                     if (isPortrait) {
-                        rectWidth = (canvasWidth) - (horizontalPadding.toPx() * 2) - (ScannerAimRectCornerPadding.toPx() * 2)
-                        rectHeight = rectWidth
-                    } else {
                         rectWidth = canvasWidth - (horizontalPadding.toPx() * 2) - (ScannerAimRectCornerPadding.toPx() * 2)
-                        rectHeight = canvasHeight - (ScannerAimRectCornerPadding.toPx() * 2)
+                        rectHeight = canvasHeight - (verticalPadding.toPx() * 2) - (ScannerAimRectCornerPadding.toPx() * 2)
+                    } else {
+                        // For phone landscape, use the full width
+                        rectWidth = canvasWidth - (ScannerAimRectCornerPadding.toPx() * 2)
+                        rectHeight = canvasHeight - (verticalPadding.toPx() * 2) - (ScannerAimRectCornerPadding.toPx() * 2)
```

Comment:
Considering that line 402 is the same as 406, maybe we could remove the duplication?

Reply:
this still needs to be addressed @ItsChaceD